lib/purchases_flutter.dart

```diff
@@ -43,13 +43,17 @@ class Purchases {
   static final Set<ReadyForPromotedProductPurchaseListener>
       _readyForPromotedProductPurchaseListeners = {};
 
+  static CustomerInfo? _lastReceivedCustomerInfo;
+
   static final _channel = const MethodChannel('purchases_flutter')
     ..setMethodCallHandler((call) async {
       switch (call.method) {
         case 'Purchases-CustomerInfoUpdated':
+          final args = Map<String, dynamic>.from(call.arguments);
+          final customerInfo = CustomerInfo.fromJson(args);
```

Comment:
I moved this outside the for loop. It didn't make sense to perform this operation on every iteration.